app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -5,17 +5,14 @@ module HasMonitoringManagerMixin
 
   def ensure_monitoring_manager
     # monitoring_manager should be defined by child classes.
-    if try(:monitoring_manager_needed?)
+    if monitoring_manager_needed? && monitoring_manager.nil?
       build_monitoring_manager(:parent_manager => self)
       monitoring_manager.name = "#{name} Monitoring Manager"
-    end
-    ensure_monitoring_manager_properties
-  end
-
-  def ensure_monitoring_manager_properties
-    if monitoring_manager
       monitoring_manager.zone_id = zone_id
       monitoring_manager.provider_region = provider_region
+    elsif !monitoring_manager_needed? && monitoring_manager.present?
+      # TODO: if someone deletes the alerts endpoint and then quickly readds it they can end up without a manager.
```

Comment:
LGTM, I don't see any easy fix, and hard fixes are out of scope.
I wonder if long-term we'll want API & UI to actually create/delete managers — and deal with manager deletion being async — rather than just creating/deleting endpoints.